server/jetty/src/main/java/io/deephaven/server/jetty/ControlledCacheResource.java

```diff
@@ -0,0 +1,122 @@
+package io.deephaven.server.jetty;
+
+import org.eclipse.jetty.util.resource.Resource;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.nio.channels.ReadableByteChannel;
+
+/**
+ * Simple wrapper around the Jetty Resource type, to grant us control over caching features. The current implementation
+ * only removes the last-modified value, but a future version could provide a "real" weak/strong etag.
+ */
+public class ControlledCacheResource extends Resource {
+    private final Resource wrapped;
+
+    public ControlledCacheResource(Resource wrapped) {
+        this.wrapped = wrapped;
+    }
+
+    @Override
+    public boolean isContainedIn(Resource r) throws MalformedURLException {
+        return wrapped.isContainedIn(r);
+    }
+
+    @Override
+    public void close() {
+        wrapped.close();
+    }
+
+    @Override
+    public boolean exists() {
+        return wrapped.exists();
+    }
+
+    @Override
+    public boolean isDirectory() {
+        return wrapped.isDirectory();
+    }
+
+    @Override
+    public long lastModified() {
+        // Always return -1, so that we don't get the build system timestamp. In theory we could return the app startup
+        // time as well, so that clients that connect don't need to revalidate quite as often, but this could have other
+        // side effects such as in load balancing with a short-lived old build against a seconds-older new build.
+        return -1;
```

Comment:
This will still be an issue with clients that have already stored the `Last-modified` header. Still get an issue with the client passing `If-Modified-Since: Fri, 01 Feb 1980 05:00:00 GMT`, I'm guessing somewhere Jetty is still checking against what the reported lastModified date is (even if it's not in a header?) and I'm still getting a 304: 

<img width="589" alt="image" src="https://user-images.githubusercontent.com/4505624/199072369-9008d660-f3b2-4375-8546-68791ed1c8de.png">

Everybody upgrading from v0.17 will still hit this issue and they'll be broken until they do a hard refresh.

Reply:
This should be fixed now, thanks for bringing it up.